sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -660,5 +715,28 @@ private static ReadOnlyMemory<byte> ReadStreamToMemory(Stream stream)
 
             return new ReadOnlyMemory<byte>(memStream.ToArray());
         }
+
+        /// <summary>
+        ///   The set of system annotations set on a message received from the
+        ///   Event Hubs service.
+        /// </summary>
+        ///
+        private struct ParsedAnnotations
+        {
+            /// <summary>The set of weakly typed annotations associated with the message.</summary>
+            public Dictionary<string, object> ServiceAnnotations;
+
+            /// <summary>The sequence number of the event associated with the message.</summary>
+            public long? SequenceNumber;
```

Comment:
In the Java Track 1, if it is unable to find this sequence number in the received message, an illegal argument exception is thrown with "sequenceNumber should always be in map." Are there cases where it is nullable?

Reply:
When you create an event to send.    The same type is used in both directions.  The expectation is that is null for outgoing events, which makes more sense than setting it to a default value.   It, quite literally is, "there is no data here."